### DISCUSSION TITLE
Strange HTTP 404 with nginx-fpm when using media file suffix in request

### DISCUSSION THREAD

**Discussion by ohader** [ddev/ddev#5214] [Q&A]
Hi folks, I experienced a strange problem with DDEV 1.22.0 when using `nginx-fpm`, which is not reproducible when using `apache-fpm`.

The PHP application (based on Laravel) accepts URLs in the form of `/[base64-encoded-payload].[arbitrary file suffix]` - only the Base64-part is relevant for the app, the file suffix is a "hint" for browsers.

The examples further down show the details of requesting the same resource with mime-type `image/svg+xml`. It seems that there is a "mime-type to file suffix" check/handling in nginx which I could not find in the default configurations. Here's a short summary of those requests:

* nginx-fpm
  + `/[base64-payload].svg` → `404` 🛑 
  + `/[base64-payload]` → `200` ✅ 
  + `/[base64-payload].bmp` → `200` ✅ 
* apache-fpm
  + `/[base64-payload].svg` → `200` ✅  
  + `/[base64-payload]` → `200` ✅ 
  + `/[base64-payload].bmp` → `200` ✅ 

Thanks in advance for any hint on this topic.

---

#### nginx-fpm `/[base64-payload].svg`

```
curl -Ik https://proxy.csp-simulator.ddev.site/aHR0cHM6Ly9pbmRpZW11c2lrLWZlc3RpdmFsLmRlL19hc3NldHMvYWY0NmYxODUzZTRlMjU5Y2JiOGViY2I4MTZlYjA0MDMvVmVjdG9ycy9pbmRpZS1sb2dvLWludmVydGVkLnN2Zw.svg
HTTP/2 404
cache-control: private, must-revalidate
content-type: image/svg+xml
date: Sun, 30 Jul 2023 09:02:28 GMT
last-modified: Sun, 25 Jun 2023 12:35:48 GMT
server: nginx
vary: Accept-Encoding
vary: Accept-Encoding
x-content-type-options: nosniff
x-fetched-uri: https://indiemusik-festival.de/_assets/af46f1853e4e259cbb8ebcb816eb0403/Vectors/indie-logo-inverted.svg
content-length: 15296
```

#### nginx-fpm `/[base64-payload]` (no file suffix)

```
curl -Ik https://proxy.csp-simulator.ddev.site/aHR0cHM6Ly9pbmRpZW11c2lrLWZlc3RpdmFsLmRlL19hc3NldHMvYWY0NmYxODUzZTRlMjU5Y2JiOGViY2I4MTZlYjA0MDMvVmVjdG9ycy9pbmRpZS1sb2dvLWludmVydGVkLnN2Zw
HTTP/2 200
cache-control: private, must-revalidate
content-type: image/svg+xml
date: Sun, 30 Jul 2023 09:03:05 GMT
last-modified: Sun, 25 Jun 2023 12:35:48 GMT
server: nginx
vary: Accept-Encoding
vary: Accept-Encoding
x-content-type-options: nosniff
x-fetched-uri: https://indiemusik-festival.de/_assets/af46f1853e4e259cbb8ebcb816eb0403/Vectors/indie-logo-inverted.svg
content-length: 15296
```

#### apache-fpm `/[base64-payload].svg`

```
curl -Ik https://proxy.csp-simulator.ddev.site/aHR0cHM6Ly9pbmRpZW11c2lrLWZlc3RpdmFsLmRlL19hc3NldHMvYWY0NmYxODUzZTRlMjU5Y2JiOGViY2I4MTZlYjA0MDMvVmVjdG9ycy9pbmRpZS1sb2dvLWludmVydGVkLnN2Zw.svg
HTTP/2 200
cache-control: private, must-revalidate
content-type: image/svg+xml
date: Sun, 30 Jul 2023 09:04:07 GMT
last-modified: Sun, 25 Jun 2023 12:35:48 GMT
server: Apache/2.4.56 (Debian)
vary: Accept-Encoding
x-content-type-options: nosniff
x-fetched-uri: https://indiemusik-festival.de/_assets/af46f1853e4e259cbb8ebcb816eb0403/Vectors/indie-logo-inverted.svg
content-length: 15296
```

**Comment by rfay** [comment]
So nice to see you here @ohader ! It's been years :)

Laravel uses this nginx config by default, https://github.com/ddev/ddev/blob/master/pkg/ddevapp/webserver_config_assets/nginx-site-laravel.conf

The Apache config is a standard one, https://github.com/ddev/ddev/blob/master/pkg/ddevapp/webserver_config_assets/apache-site-php.conf

Often folks using Apache are counting on the behavior of a project's various .htaccess files, which we don't see here, but you might be getting those when using apache.

My bet is that either it's that or the Laravel Nginx config needs to be improved. You may have a suggestion from the Laravel community about how to improve the nginx config, or you may find something in your Apache config.

You also might want to try other project types, like typo3, that use different default nginx config. The various configs are in https://github.com/ddev/ddev/tree/master/pkg/ddevapp/webserver_config_assets
